src/rt/aaA.d

```diff
@@ -480,7 +480,7 @@ inout(ArrayRet_t) _aaKeys(inout AA aa, in size_t keysize, const TypeInfo tiKeyAr
     if (!len)
         return null;
 
-    void[] res = _d_newarrayU(tiKeyArray, len);
+    void* res = _d_newarrayU(tiKeyArray, len).ptr;
```

Comment:
Yeah, it crept in because there is a range error when building a debug version of druntime. I think return `void*` from `_d_newarrayU` would be less confusing. I'll make another pull request for this.